### PR TITLE
Use Semaphore for CI

### PR DESCRIPTION
**Pull request by cdmistman**

### CODE REVIEW

**Comment by cdmistman** [comment]
semaphore works, i asked for circleci to be removed so once it's removed we won't get the ugly `x`s anymore :)

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            - cache restore goreleaser
+            - goreleaser --release-notes=<(scripts/extract-changelog.py)
+        - name: snapcraft
+          commands:
+            - >-
+              snapcraft login --with <(
+                echo "$SNAPCRAFT_LOGIN_FILE" |
+                base64 --decode --ignore-garbage
+              )
+            - scripts/upload-snap.bash
```
Comment (cdmistman) [inline .semaphore/semaphore.yml:113]: haven't tested, but if we ever do another release we can figure that out then :) things are mostly implemented

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        - name: nix
+          commands:
+            - cache restore nix-store
+            - nix build
+            - ./result/bin/upm --version
+            - cache store nix-store /nix
+        # TODO: get this working again
+        # - name: docker
+        #   commands:
+        #     - make light full
```
Comment (cdmistman) [inline .semaphore/semaphore.yml:71]: some package we depend on in the docker images doesn't have the right version when building the image, i figured it wasn't important enough to keep diagnosing

Comment (masad-frost) [reply .semaphore/semaphore.yml:71]: yeah let's skip the docker builds

File: .semaphore/semaphore.yml (hunk 1 above)
Comment (masad-frost) [inline .semaphore/semaphore.yml:113]: I'd also skip snap

**Review by masad-frost — APPROVED** [review]

**Pull request merged**